src/main/java/de/zalando/aruha/nakadi/repository/db/EventTypeCache.java

```diff
@@ -0,0 +1,140 @@
+import org.apache.zookeeper.KeeperException;
+
+import java.util.concurrent.ExecutionException;
+
+public class EventTypeCache {
+
+    public final String ZKNODE_PATH = "/nakadi/event_types";
+
+    private final LoadingCache<String, EventType> cache;
+    private final PathChildrenCache cacheSync;
+    private final CuratorFramework zkClient;
+
+    public EventTypeCache(EventTypeRepository dbRepo, CuratorFramework zkClient) throws Exception {
+        initParentCacheZNode(zkClient);
+
+        this.zkClient = zkClient;
+        this.cache = setupInMemoryCache(dbRepo);
+        this.cacheSync = setupCacheSync(zkClient);
+    }
+
+    public void updated(String name) {
+        try {
+            String path = getZNodePath(name);
+            zkClient.setData().forPath(path, new byte[0]);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public EventType get(String name) throws NoSuchEventTypeException {
+        try {
+            return cache.get(name);
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof NoSuchEventTypeException) {
+                NoSuchEventTypeException noSuchEventTypeException = (NoSuchEventTypeException) e.getCause();
+                throw noSuchEventTypeException;
+            } else {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public void created(EventType eventType) {
+        try {
+            String path = getZNodePath(eventType.getName());
+            zkClient
+                    .create()
+                    .creatingParentsIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath(path, new byte[0]);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void removed(String name) {
+        try {
+            String path = ZKPaths.makePath(ZKNODE_PATH, name);
+            zkClient.delete().forPath(path);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void initParentCacheZNode(CuratorFramework zkClient) throws Exception {
+        try {
+            zkClient
+                    .create()
+                    .creatingParentsIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath(ZKNODE_PATH);
+        } catch (KeeperException.NodeExistsException e) {
+            System.out.println("NODE ALREADY EXISTS");
+        }
+    }
+
+    private void addCacheChangeListener(final LoadingCache<String, EventType> cache, PathChildrenCache cacheSync) {
+        PathChildrenCacheListener listener = new PathChildrenCacheListener() {
+            @Override
+            public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) throws Exception {
+                switch ( event.getType() ) {
+                    case CHILD_UPDATED: {
+                        invalidateCacheKey(event);
```

Comment:
When something changes, just invalidate the cache and waits for the nest request to reload it, if necessary.